src/components/scenes/newAccount/NewAccountReviewScene.tsx

```diff
@@ -93,17 +93,16 @@ export const NewAccountReviewScene = (props: NewAccountReviewProps) => {
   const dispatch = useDispatch()
   const { onLogin } = useImports()
 
-  const handleNext = useHandler(() => {
-    const lightAccount = route.params.username == null
-    logEvent(`Signup_Review_Done`, { lightAccount })
+  if (onLogin == null)
+    throw new Error('NewAccountReviewScene: onLogin not found')
```

Comment:
Please don't throw at the top level of a component. This is really hard to debug.

The previous structure, `if (onLogin != null) onLogin()` was better anyhow, since if you forget to provide a callback, everything else would work like normal. You just never get called.

Imagine if a button component crashed when you pressed it, because you forgot to provide an `onPress`! Buttons don't work this way. They still handle the press and show the animation, etc.. They just never tell you about it, because you never signed up to hear about it. The same principle applies here.

Reply:
The idea was in the described case it's _not_ working like normal - there's never a time in which we would be OK with this situation happening. This assertion is communicating that the configuration is invalid, and screaming to fix it before it leaves the developer.

Reply:
I think your button example works for a generic component, and of course assertions don't make sense in that case because it is intended design that the component supports the scenario in which an onPress isn't passed.

I'd argue that it's harder to debug by just ignoring the unintended setups. I have had situations where something I'm unfamiliar with is just inexplicably misbehaving, requiring me to look into it further. With an assertion, it's obvious.

Maybe move the assertion down into the next press then?